src/codegen/expression.rs

```diff
@@ -237,13 +238,21 @@ pub fn expression(
         }
         ast::Expression::Assign(_, _, left, right) => {
             // If we reach this condition, the assignment is inside an expression.
+
             if let Some(function) = func {
                 if should_remove_assignment(ns, left, function, opt) {
                     return expression(right, cfg, contract_no, func, ns, vartab, opt);
                 }
             }
 
-            let cfg_right = expression(right, cfg, contract_no, func, ns, vartab, opt);
+            let mut cfg_right = expression(right, cfg, contract_no, func, ns, vartab, opt);
+
+            // If an assignment where the left hand side is an array, call a helper function that updates the temp variable.
+            if let ast::Expression::Variable(_, Type::Array(..), pos) = &**left {
+                // If cfg_right is an AllocDynamicArray(_,_,size,_), update it such that it becomes AllocDynamicArray(_,_,temp_var,_) to avoid repetitive expressions in the cfg.
+                cfg_right = handle_array_assign(cfg_right.clone(), cfg, vartab, pos);
```

Comment:
```suggestion
                cfg_right = handle_array_assign(cfg_right, cfg, vartab, pos);
```
There is no need to clone here, because you are regenerating `cfg_right`, so a move suffices.